### DISCUSSION TITLE
Is it possible to turn off encryption?

### DISCUSSION THREAD

**Discussion by Beatle95** [microsoft/msquic#3196] [Q&A] [answered]
I want to get maximum speed and my thought was: maybe if I turn off encryption I will get even better results?
So my question is: is it possible to turn off encryption for MsQuic?

P.S. I know QUIC is encrypted by default, but still, just to be sure, maybe this is possible?

**Comment by nibanks** [comment, marked as answer]
If you have a network that can handle multiple gigabit per second, then yes, encryption may be holding you back. We don't recommend it's usage, but we support [this](https://github.com/microsoft/msquic/blob/main/src/inc/msquic.h#L832), which must be configured on both client and server for it to work.

**Reply by Beatle95** [reply]
Thank you!

**Reply by Beatle95** [reply]
Btw, I can't find information about QUIC_PARAM_CONN_DISABLE_1RTT_ENCRYPTION parameter.
I assume that on client side I set it to true right after ConnectionOpen, looks like it works fine. But when shoud I do this on server side? Inside listener callback on QUIC_LISTENER_EVENT_NEW_CONNECTION event?

P.S. I am asking because at this moment I get some strange runtime errors, if I set it in QUIC_LISTENER_EVENT_NEW_CONNECTION event right before SetCallbackHandler call.

**Reply by Beatle95** [reply]
I have discovered, that the problem was at the client side. I have accidentally set QUIC_PARAM_CONN_DISABLE_1RTT_ENCRYPTION to false there.
